packages/asset-swapper/src/utils/market_operation_utils/comparison_price.ts

```diff
@@ -4,56 +4,42 @@ import * as _ from 'lodash';
  * So the RFQT MM should aim for a higher price
- * @param optimizerResult the output of the optimizer, contains the best orders
+ * @param adjustedRate the adjusted rate (accounting for fees) from the optimizer, maker/taker
  * @param amount the amount specified by the client
  * @param marketSideLiquidity the results from querying liquidity sources
+ * @param feeSchedule the fee schedule passed to the Optimizer
  * @return ComparisonPrice object with the prices for RFQ MMs to beat
  */
 export function getComparisonPrices(
-    optimizerResult: OptimizerResult,
+    adjustedRate: BigNumber,
     amount: BigNumber,
     marketSideLiquidity: MarketSideLiquidity,
+    feeSchedule: FeeSchedule,
 ): ComparisonPrice {
-    // the adjusted rate is defined as maker/taker
-    // input is the taker token for sells, input is the maker token for buys
-    const makerTakerOptimalRate = optimizerResult.adjustedRate;
-    const takerMakerOptimalRate = optimizerResult.adjustedRate.pow(-1);
-
-    // fees for a native order
-    const fees = optimizerResult.exchangeProxyOverhead(SOURCE_FLAGS.Native);
-
     // Calc native order fee penalty in output unit (maker units for sells, taker unit for buys)
     const feePenalty = !marketSideLiquidity.ethToOutputRate.isZero()
-        ? marketSideLiquidity.ethToOutputRate.times(fees)
+        ? marketSideLiquidity.ethToOutputRate.times((feeSchedule[ERC20BridgeSource.Native] as FeeEstimate)(undefined))
         : // if it's a sell, the input token is the taker token
           marketSideLiquidity.ethToInputRate
-              .times(fees)
-              .times(marketSideLiquidity.side === MarketOperation.Sell ? makerTakerOptimalRate : takerMakerOptimalRate);
-
-    let wholeOrder: BigNumber | undefined;
+              .times((feeSchedule[ERC20BridgeSource.Native] as FeeEstimate)(undefined))
+              .times(marketSideLiquidity.side === MarketOperation.Sell ? adjustedRate : adjustedRate.pow(-1));
 
-    let orderMakerAmount: BigNumber;
-    let orderTakerAmount: BigNumber;
-    if (marketSideLiquidity.side === MarketOperation.Sell) {
-        orderTakerAmount = amount;
-
-        orderMakerAmount = makerTakerOptimalRate.times(orderTakerAmount).plus(feePenalty);
-    } else if (marketSideLiquidity.side === MarketOperation.Buy) {
-        orderMakerAmount = amount;
-
-        orderTakerAmount = takerMakerOptimalRate.times(orderMakerAmount).minus(feePenalty);
-    } else {
-        throw new Error(`Unexpected marketOperation ${marketSideLiquidity.side}`);
-    }
+    // the adjusted rate is defined as maker/taker
+    // input is the taker token for sells, input is the maker token for buys
+    const orderMakerAmount =
+        marketSideLiquidity.side === MarketOperation.Sell ? adjustedRate.times(amount).plus(feePenalty) : amount;
+    const orderTakerAmount =
+        marketSideLiquidity.side === MarketOperation.Sell ? amount : amount.dividedBy(adjustedRate).minus(feePenalty);
```

Comment:
OK so you are not doing the `^ -1` any more? or is there some other change?

Reply:
I simplified this block of code to just use this ternary logic. I condensed 7-8 lines or so into these two. Functionally they should be the same.